### PR TITLE
add prep-svn.mk for pull source from subversion repositories

### PR DESCRIPTION
**Pull request by cgrzemba**
this are from solaris-userland to use svn repo for pulling component sources. Usefull for projects which do not provide tag versions

### CODE REVIEW

File: components/network/vpnc/Makefile [hunk 1]
```diff
@@ -15,21 +15,26 @@
 include ../../../make-rules/shared-macros.mk
 
 COMPONENT_NAME=		vpnc
-COMPONENT_VERSION=	0.5.3
```
Comment (pyhalov) [inline components/network/vpnc/Makefile]: Can we preserve COMPONENT_VERSION, bump COMPONENT_REVISION and set HUMAN_VERSION to something like $(COMPONENT_VERSION).r550 ?

**Comment by pyhalov** [comment]
please, remove merge commit from PR (you can use rebase)

File: components/network/vpnc/Makefile (hunk 1 above)
Comment (cgrzemba) [reply components/network/vpnc/Makefile]: so we get

oi-userland network/vpnc                                    0.5.3-2017.0.0.2:20170221T154918Z

as package version?

Comment (jimklimov) [reply components/network/vpnc/Makefile]: With REVISION(==1 for the first bump) in place, AFAIK you get something like `...0.5.3.1` which is above any 0.5.3 or 0.5.3.0 but under any 0.5.4.

Comment (pyhalov) [reply components/network/vpnc/Makefile]: @cgrzemba : yes

**Comment by cgrzemba** [comment]
things are done

File: components/network/vpnc/Makefile [hunk 2]
```diff
@@ -16,20 +16,25 @@ include ../../../make-rules/shared-macros.mk
 
 COMPONENT_NAME=		vpnc
 COMPONENT_VERSION=	0.5.3
-COMPONENT_REVISION=	1
+COMPONENT_REVISION=	0
```
Comment (pyhalov) [inline components/network/vpnc/Makefile]: Why???

File: components/network/vpnc/Makefile [hunk 3]
```diff
@@ -16,20 +16,25 @@ include ../../../make-rules/shared-macros.mk
...
+# COMPONENT_ARCHIVE_HASH=	\
 	sha256:46cea3bd02f207c62c7c6f2f22133382602baeda1dc320747809e94881414884
 COMPONENT_LICENSE=		GPLv2,BSD-2clause
 COMPONENT_LICENSE_FILE= COPYING
 COMPONENT_SUMMARY=	An IPSec-like VPN client compatible with Cisco EasyVPN
 COMPONENT_DESCRIPTION=	An IPSec-like (over TUN/TAP) VPN client compatible with Cisco 3000 VPN Concentrators and EasyVPN
 
+SVN_REPO = http://svn.unix-ag.uni-kl.de/vpnc/trunk
+SVN_REV = 550
+SVN_HASH = NOHASH
```
Comment (pyhalov) [inline components/network/vpnc/Makefile]: Shouldn't it be SVN_HASH= for proper prep-svn.mk work?
Also note that hash should be populated before merging (we can upload tarball to server's USERLAND_ARCHIVES)

File: components/network/vpnc/Makefile [hunk 4]
```diff
@@ -16,20 +16,25 @@ include ../../../make-rules/shared-macros.mk
...
 COMPONENT_ARCHIVE=	$(COMPONENT_NAME)-$(COMPONENT_VERSION).tar.gz
 COMPONENT_FMRI=		network/$(COMPONENT_NAME)
 COMPONENT_CLASSIFICATION=	Applications/Internet
-COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
+# COMPONENT_SRC=		$(COMPONENT_NAME)-$(COMPONENT_VERSION)
 COMPONENT_PROJECT_URL=	https://www.unix-ag.uni-kl.de/~massar/vpnc/
-COMPONENT_ARCHIVE_URL=	https://www.unix-ag.uni-kl.de/~massar/vpnc/$(COMPONENT_ARCHIVE)
-COMPONENT_ARCHIVE_HASH=	\
+# COMPONENT_ARCHIVE_URL=	https://www.unix-ag.uni-kl.de/~massar/vpnc/$(COMPONENT_ARCHIVE)
+# COMPONENT_ARCHIVE_HASH=	\
```
Comment (pyhalov) [inline components/network/vpnc/Makefile]: please, remove commented out lines

**Comment by xen0l** [comment]
@pyhalov can we go and merge this?

**Comment by pyhalov** [comment]
At least without merge commit, which we see here.
Also note that SVN_HASH should be manually updated after component is built, otherwise it will be republished each time.

**Comment by cgrzemba** [comment]
The merge commits are removed.

**Comment by pyhalov** [comment]
please, remove exiv2 update commit from this PR

**Comment by cgrzemba** [comment]
removed.

**Comment by pyhalov** [comment]
Looks good, but after this is merged one should to set SVN_HASH to actual value (as it will be changed on build). Will look at it in 1-2days.

**Pull request merged**